### PR TITLE
ACM-30436 Argo cluster-scoped topology status

### PR DESCRIPTION
**Pull request by KevinFCormier**
# 📝 Summary

**Ticket Summary (Title):**  
AppSet cluster scoped node status incorrect in the topology node details panel

**Ticket Link:**  
https://redhat.atlassian.net/browse/ACM-30436

**Type of Change:**  

- [x] 🐞 Bug Fix  
- [ ] ✨ Feature  
- [ ] 🔧 Refactor
- [ ] 💸 Tech Debt
- [ ] 🧪 Test-related  
- [ ] 📄 Docs

---

### General

- [x] PR title follows the convention (e.g. `ACM-12340 Fix bug with...`)
- [x] Code builds and runs locally without errors
- [x] No console logs, commented-out code, or unnecessary files
- [x] All commits are meaningful and well-labeled
- [x] All new display strings are externalized for localization (English only)
- [x] *(Nice to have)* JSDoc comments added for new functions and interfaces

#### If Bugfix

- [ ] Root cause and fix summary are documented in the ticket (for future reference / errata)
- [x] Fix tested thoroughly and resolves the issue
- [x] Test(s) added to prevent regression

---

## Before

https://github.com/user-attachments/assets/3bac1d54-8ec1-4c13-bd84-28ddb0c39beb

## After

https://github.com/user-attachments/assets/626ce5a1-9b1e-40f1-b4b5-18a5541d071a

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed resource display for pull-model ApplicationSets when remote status data is missing and avoided unnecessary remote fetches when status is already present.
  * Prevented duplication of cluster-scoped resources across clusters.

* **Improvements**
  * Topology now enriches and merges remote application resource lists from managed clusters, maps remote health into topology statuses, and aggregates cluster info for cluster-scoped nodes.
  * Search queries for non-namespace-scoped resources are explicitly scoped to target clusters.

* **Tests**
  * Added tests for pull-model fetch/merge flows, uniform vs non-uniform sources, deduplication, and error/edge cases.

### CODE REVIEW

File: frontend/src/routes/Applications/ApplicationDetails/ApplicationTopology/model/topologyAppSet.ts [hunk 1]
```diff
@@ -520,12 +648,12 @@ function processResources(
...
         raw,
-        clustersNames: parentClusterNames,
+        clustersNames: nodeClusters,
         parent: {
           clusterId: parentId,
         },
         resources: deployableResources,
-        resourceCount: resourceCount || parentClusterNames.length,
+        resourceCount: resourceCount || nodeClusters.length,
       },
```
[the remark this answers is not in the document]
Comment (KevinFCormier) [reply frontend/src/routes/Applications/ApplicationDetails/ApplicationTopology/model/topologyAppSet.ts:700]: - resourceCount comes from processMultiples and is only set when grouping multiple resources of the same kind (>5 total). For individual resources it's undefined, so resourceCount || nodeClusters.length correctly falls back to nodeClusters.length.
- After dedup, a single cluster-scoped resource has _targetClusters with all clusters, so nodeClusters.length gives the correct cluster count.
- When processMultiples does set resourceCount (grouped node representing N resources of the same kind), that count should take priority over nodeClusters.length since it represents the number of resources in the group, not the number of clusters.
- The proposed change (clustersNames?.length ?? resourceCount ?? nodeClusters.length) would be incorrect for grouped nodes — it would show the cluster count instead of the resource group count.

File: frontend/src/routes/Applications/ApplicationDetails/ApplicationTopology/model/topologyAppSet.ts [hunk 2]
```diff
@@ -547,6 +753,38 @@ function processResources(
...
+      const existing = clusterScoped.get(key)
+      if (existing) {
+        const clusters = (existing as any)._targetClusters || [existing.cluster]
+        if (resource.cluster && !clusters.includes(resource.cluster)) {
+          clusters.push(resource.cluster)
+        }
+        ;(existing as any)._targetClusters = clusters
+      } else {
+        const entry = { ...resource, _targetClusters: resource.cluster ? [resource.cluster] : [] }
+        clusterScoped.set(key, entry)
```
[the remark this answers is not in the document]
Comment (KevinFCormier) [reply frontend/src/routes/Applications/ApplicationDetails/ApplicationTopology/model/topologyAppSet.ts:745]: The current code is correct because an empty _targetClusters array can only occur when existing.cluster is also falsy — line 779 ensures they're always in sync (_targetClusters: resource.cluster ? [resource.cluster] : []). The proposed fallback to [existing.cluster] would produce [undefined], which is worse than []. No change needed.

**Review by fxiang1 — APPROVED** [review]
Looks great Kevin! Thanks for fixing this! Looks like it wasn't an easy fix :)

**Comment by jeswanke** [comment]
cool! /lgtm

**Review by jeswanke — APPROVED** [review]
